sunkit_instruments/suvi/io.py

```diff
@@ -0,0 +1,320 @@
+import os
+
+import h5py
+import numpy
+
+from astropy import units as u
+from astropy.io import fits
+from astropy.time import Time
+
+from sunkit_instruments.suvi import fix_L1b_header
+
+__all__ = ["read_suvi"]
+
+def read_suvi(input_filename, return_DQF=False, return_header_only=False):
+    """
+    Read a SUVI L1b FITS or netCDF file or a L2 HDR composite FITS file.
+    Return data and header, optionally the data quality flag array (DQF)
+    for L1b files. For SUVI L1b FITS files, the broken FITS header is
+    fixed automatically (broken because of the wrong implementation of
+    the CONTINUE convention). This read function is intented to provide
+    a consistent file interface for FITS and netCDF, L1b and L2.
+
+    .. note::
+        The type of file is determined by pattern matching in the
+        filenames, e.g. "-L1b-Fe171" for a 171 L1b file and "-l2-ci171"
+        for a 171 L2 HDR composite. If those patterns are not found
+        in the filename, the files will not be recognized.
+
+    .. note::
+        If input_filename is an L1b netCDF file, the information from
+        the netCDF file is transformed into a FITS header.
+
+    Parameters
+    ----------
+    input_filename: `str`
+        File to read.
+
+    return_DQF: `bool`, optional. Default: False.
+        If True, also returns the data quality flag array (DQF, only for L1b files).
+
+    return_header_only: `bool`, optional. Default: False.
+        If True, does not return the data array (but optionally the DQF).
+
+    Returns
+    -------
+    header: `astropy.io.fits.header.Header`
+        header, if return_header_only is True.
+
+    header, data: `astropy.io.fits.header.Header`, `~numpy.ndarray`
+        header and data.
+
+    header, dqf: `astropy.io.fits.header.Header`, `~numpy.ndarray`
+        header and data quality flags, if both return_header_only and return_DQF
+        are True.
+
+    header, data, dqf: `astropy.io.fits.header.Header`, `~numpy.ndarray`, `~numpy.ndarray`
+        header, data, and data quality flags if return_DQF is True.
```

Comment:
What was your reason for allowing this range of outputs back? 

Normally in any of our IO returns, we would want to return a fixed set of variables either, so it can be passed into a sunpy.map.Map.

Reply:
Mostly so that the user can request the exact thing he/she wants. If I return the DQF every time, or the data when it is not really needed, it is unnecessary io.

Reply:
(also because the L2 composites do not have a DQF)

Reply:
Actually, they do, never mind... I would like to keep the flexibility, but if instead you just want me to always return header, data, dqf, let me know an I can do that. I don't really see a good reason though why one would return things the user hasn't requested.

Reply:
Never mind, I am just returning None if if was not requested, so it will always be three arguments that are being returned. The user can do `header, data, _ = read_suvi(...)` then for example.